File: archive/java/muse/algorithms/sorting/SelectionSortTest.java
import muse.algorithms.sorting.SelectionSort;
import muse.util.SequenceBuilder;
import muse.util.Sequences;
import muse.util.TestRunner;

public class SelectionSortTest {

  public static boolean testSelectionSort() {
    int size = 32768;

    Integer[] arr = new Integer[size];
    SequenceBuilder.packRandom(arr);

    int checksum = Sequences.parityChecksum(arr);

    SelectionSort.sort(arr);

    if (Sequences.parityChecksum(arr) != checksum) {
      return false;
    }

    return Sequences.isSorted(arr);
  }

  public static void main(String[] args) {
    TestRunner.pick(SelectionSortTest::testSelectionSort);
  }
}

File: archive/nodejs/muse/algorithms/sorting/SelectionSortTest.js
import * as SequenceBuilder from '../../util/SequenceBuilder.js';
import * as Sequences from '../../util/Sequences.js';
import * as TestRunner from '../../util/TestRunner.js';
import * as SelectionSort from './SelectionSort.js';

const testSelectionSort = () => {
  const size = 32768;

  const arr = new Array(size);
  SequenceBuilder.packRandom(arr);

  const checksum = Sequences.parityChecksum(arr);

  SelectionSort.sort(arr);

  if (Sequences.parityChecksum(arr) !== checksum) {
    return false;
  }

  return Sequences.isSorted(arr);
};

const main = () => {
  TestRunner.pick(testSelectionSort);
};

main();
